### ISSUE TITLE
Include "block contact" and "disable notifications" in "contact details" view 

### ISSUE THREAD

**Issue by moshpirit** [siacs/Conversations#889]
The menu it's full of options and it can be a little frustrating if we're searching anything. That's why I've thought this solution.

BTW, IMO adding a "search in the record" could be a good option to add.

**Comment by iNPUTmice** [comment]
You are correct that menu is too full and this has bothered for a while as well. However the problem is that if we just move block and mute into contact details you won't be able to block/mute contacts any more without adding them to your contact list. (The contact details will only show up if the contact has been added. It will show an add dialog otherwise)

One work around would be to add the block option to the 'Add to roster dialog' and just don't offer the mute option. But this way you woudn't be able to mute Conversations any longer where the contact isn't in the roster (only block) but maybe that would be acceptable?

**Comment by SamWhited** [comment]
I've got a branch which adds the menu items to the details views with some caveats: Block is only in the contacts detail view, and mute is only in the conference details view. My rationale here was that "block" is an action on a contact, and "mute" is an action on a conversation. Conference details describe a conversation, and contact details describe a contact.

This does not solve the larger problem of the contact menu item getting cluttered up. Being able to block a contact right from the conversation is nice, but I'm not sure if it's strictly necessary. There could still be a "contact details" for contacts not in the roster, and you could still unblock from the conversation, the details view, or the block list dialog (if you've closed the conversation and can't get back to the details view). This basically keeps the status quo (if you close a conversation with a contact who's not in your roster, you no longer have a target to block them).

As for mute/unmute, I think those are probably used a bit more often [citation needed] and should remain in the conversation as well as being added to the conference details view. My intuition says that mute/unmute is something you'd want quick access too. Maybe giving it an icon and allowing it to flow into the action bar if there's room would be a good temporary stop gap measure (albeit, one that only helps on large phones)? I'm not sure if it's a commonly used enough action for that to be a good idea or not.

**Comment by moshpirit** [comment]
>However the problem is that if we just move block and mute into contact details you won't be able to block/mute contacts any more without adding them to your contact list. 

Oh, Ok, got it :(
About the "roster dialog" it's ok for me, maybe it could be the solution to any other option. Thanks :)

>My rationale here was that "block" is an action on a contact, and "mute" is an action on a conversation. Conference details describe a conversation, and contact details describe a contact (the branch isn't really a solution to this issue, I just agreed that parts of the idea were sound and should probably be implemented either way).

It's sounds great for me too!! 

Yes, maybe an "advanced options" or "more options" could be great.

**Comment by jkufner** [comment]
Can't we have contact details without roster?

When contact is not in roster, it should be mentioned in top of the
contact detail, and some data will not be available (since they are
loaded from roster), but otherwise, I see no problem with that.

**Comment by cultixate** [comment]
Please consider expanding ticket to include full support for XEP-0016 (Privacy Lists).

**Comment by SamWhited** [comment]
@cultixate This is a UI discussion; I don't think privacy lists are really relavant here. Also of note, there's an unresolved (though stalled, at the moment) discussion happening on the XMPP standards mailing list about resolving issues between privacy lists and the blocking command. One solution that has been proposed is to deprecate privacy lists in favor of JUST having the blocking command.

The start of the thread is [here](http://mail.jabber.org/pipermail/standards/2014-December/029426.html).

**Comment by iNPUTmice** [comment]
> Can't we have contact details without roster? 

this has just been implemented. Block/unblock contact as a result has been moved out.

mute/unmute is a something specific to a Conversations and thus can not / should not be moved out.

For the moment I don't see any other menu items that can be moved away.

I might get rid of the Invite Contact thing in favor of a proper Create Conference Dialog for version 1.1 but that's out of scope for this issue.
